src/plots/cartesian/set_convert.js

```diff
@@ -406,29 +406,33 @@ module.exports = function setConvert(ax, fullLayout) {
             arrayIn = trace[axLetter];
             len = trace._length || arrayIn.length;
 
-            if(ax.type === 'linear' && Lib.isTypedArray(arrayIn) && arrayIn.subarray) {
-                arrayOut = arrayIn.subarray(0, len);
-            } else {
-                arrayOut = new Array(len);
-
-                for(i = 0; i < len; i++) {
-                    arrayOut[i] = ax.d2c(arrayIn[i], 0, cal);
+            if(Lib.isTypedArray(arrayIn)) {
+                if(len === arrayIn.length) {
+                    return arrayIn;
+                } else if(arrayIn.subarray) {
+                    return arrayIn.subarray(0, len);
```

Comment:
This should work for linear and log axes, but what happens if you feed numeric data to a non-numeric (date or category) axis? Both of those could be real use cases, but both *also* alter the input numbers, so I think they need to bail out to the `d2c` block below.

Reply:
Done and :lock: in 6dd2f69e1489982646a36fc488e9b23b5f5f315d